### DISCUSSION TITLE
Is Milvus standalone version production ready?

### DISCUSSION THREAD

**Discussion by egolearner** [milvus-io/milvus#21639] [Q&A and General discussion] [answered]
Can we deploy standalone version for production use? Or should we deploy cluster version instead?

Follow up question if standalone version is production ready
1. Are all deploy types(Helm/Operator/Docker Compose) suitable for production use?
2. As data grows, we may reach standalone version limit. Is there a tool to migrate data from standalone version to cluster version?

**Comment by yhmo** [comment, marked as answer]
The standalone version is not recommended for production.
The cluster version installed by Helm is recommended.

You can use milvus backup tool to migrate data from standalone to cluster: https://github.com/zilliztech/milvus-backup
Steps:
1. keep the standalone instance running
2. use the backup tool to extract data from the standalone instance
3. start a cluster service installed by Helm
4. import the data into the cluster service by the backup tool

Please read the README of the backup tool to learn how to use it.

**Reply by egolearner** [reply]
Thanks a lot.

**Reply by asmaier** [reply]
I'm confused. In this blog post Milvus Standalone is marked as production ready: https://milvus.io/blog/embedded-milvus.md

**Reply by yhmo** [reply]
> I'm confused. In this blog post Milvus Standalone is marked as production ready: https://milvus.io/blog/embedded-milvus.md

Just "not recommended".
You can use standalone in production, but it is not scalable. A cluster is preferred.
